stage1/init/common/mount.go

```diff
@@ -294,8 +296,29 @@ func AppAddMounts(p *stage1commontypes.Pod, ra *schema.RuntimeApp, enterCmd []st
 		os.Exit(254)
 	}
 
+	absRoot, err := filepath.Abs(p.Root)
+	if err != nil {
+		log.FatalE("could not determine pod's absolute path", err)
+	}
+
+	appRootfs := common.AppRootfsPath(absRoot, ra.Name)
+
 	for _, m := range mounts {
-		AppAddOneMount(p, ra, m.Volume.Source, m.Mount.Path, m.ReadOnly, enterCmd)
+		shPath := filepath.Join(common.SharedVolumesPath(p.Root), m.Volume.Name.String())
+		mntPath, err := EvaluateSymlinksInsideApp(appRootfs, m.Mount.Path)
```

Comment:
Add a TODO similar to https://github.com/coreos/rkt/blob/a252c4c3078f746eca2739f3560428aa59b89da2/stage1/init/common/pod.go#L501-L504 if that is relevant here as well.

I'm not sure it is given this isn't so closely related to nspawn, but if not I'd also like a comment explaining why this is needed